akka-javasdk/src/main/java/akka/javasdk/agent/CoreMemory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import java.util.List;
+
+/**
+ * Interface for managing conversation history between users and AI models, using CoreMemory
+ * provides functionality to store, retrieve, and manage messages exchanged during conversations in
+ * an agent system.
+ */
+public interface CoreMemory {
+
+  /**
+   * Adds a user message to the conversation history for the specified session.
+   *
+   * @param componentId The unique identifier for the component type that added the message
+   * @param sessionId The unique identifier for the conversation session
+   * @param message The content of the user message to add
+   */
+  void addUserMessage(String componentId, String sessionId, String message);
```

Comment:
Since the purpose of this interface is to have custom implementations of the memory used by an Agent I don't think we need to define anything more than what the agent implementation requires, which is addInteraction and getFullHistory.

Follow up would be to remove addUserMessage and addAiMessage, when we can use addInteraction.

Reply:
When I was building the ask akka sample, one thing that became clear to me when trying it out, is that we only want to add the user message if we have an AI message. 

That's why I introduced there a single `addExchange` method. 

The call to the AI might fail and if that happens, we don't want to save the user message.

Reply:
yes, that is in the follow up plan